src/compiler/moduleNameResolver.ts

```diff
@@ -761,22 +778,13 @@ namespace ts {
     }
 
     function tryLoadModuleUsingBaseUrl(extensions: Extensions, moduleName: string, loader: ResolutionKindSpecificLoader, state: ModuleResolutionState): Resolved | undefined {
-        const { baseUrl, paths } = state.compilerOptions;
+        const { baseUrl } = state.compilerOptions;
         if (!baseUrl) {
             return undefined;
         }
         if (state.traceEnabled) {
             trace(state.host, Diagnostics.baseUrl_option_is_set_to_0_using_this_value_to_resolve_non_relative_module_name_1, baseUrl, moduleName);
         }
-        if (paths) {

```

Comment:
I think this should be pulled out into separate function instead of inlining. Its easier to read.
We also need trace above this line about using baseUrl in that.

Reply:
> I think this should be pulled out into separate function instead of inlining.

I thought about that. I however couldn't think of a name. The name `tryLoadModuleUsingPaths` is already taken. What do you suggest?

> We also need trace above this line about using baseUrl in that.

The trace `baseUrl_option_is_set_to_0_using_this_value_to_resolve_non_relative_module_name_1` is not exactly correct for what is happening. And since we have the trace `paths_option_is_specified_looking_for_a_pattern_to_match_module_name_0` I thought that would be enough. But if you really want it there I could of course add it back in.

Reply:
Yes please add that back in there since its used in there anyways.